jdr/WFLY-13539-galleon_layer_jdr_support.adoc

```diff
@@ -0,0 +1,131 @@
+-p,--port <arg>       port that the management api is bound to. (default: 9990)
+-s,--protocol <arg>   Protocol that is used to connect. Can be remote, http or https (default: http)
+----
+
+The availability of a management interface is optional. If there is not a management interface available, the `jdr.{sh,bat,ps1}` script is still able to extract some information invoking the embedded server internally.
+
+The goal of this feature is to supply a https://docs.wildfly.org/galleon/#_layers[Galleon layer] to give support to the JDR subsystem.
+
+The following is a configuration example of the WildFly bootable jar maven plugin to provision a server with the `jdr` layer and the `core-tools` layer, which brings in the `jboss-cli.sh` script that can be used to connect to the server via the command line to generate a diagnostic report through the jdr subsystem:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-jar-maven-plugin</artifactId>
+    <configuration>
+        <feature-pack-location>wildfly@maven(org.jboss.universe:community-universe)#${version.wildfly}</feature-pack-location>
+        <layers>
+            <layer>jdr</layer>
+            <layer>core-tools</layer>
+        </layers>
+    </configuration>
+    <executions>
+        <execution>
+            <goals>
+                <goal>package</goal>
+            </goals>
+        </execution>
+    </executions>
+</plugin>
+----
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13539[WFLY-13539 Add a Galleon layer for JBoss Diagnostic Reporting]
+* https://issues.redhat.com/browse/EAP7-1528[EAP7-1528 Add a Galleon layer for JBoss Diagnostic Reporting]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFGP-177[WFGP-177 Create a Galleon package for the JBoss Diagnostic Reporting bin scripts]
+
+=== Dev Contacts
+
+* mailto:brian.stansberry@redhat.com[Brian Stansberry]
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+https://github.com/wildfly/wildfly[WildFly], https://github.com/wildfly/galleon-plugins[WildFly Galleon Plugins]
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+* The galleon layer must provide the provisioning of the JDR subsystem.
+* The provisioning tool will be responsible for providing the bash scripts to generate the diagnosis information - i.e. `jdr.{sh,bat,ps1}` - and make those available in the following filesystem location: `$WFLY_HOME/bin/`.
+* The galleon layer will depend optionally on `secure-management` layer. This layer will allow the users to configure a secure management interface to connect to the server via `jdr.sh`. If a user wants to connect via CLI, they have to add `core-tools` layer to their provision configuration.
```

Comment:
```
The galleon layer will depend optionally on `secure-management` layer. This layer will allow the users to configure a secure management interface to connect to the server via `jdr.sh`. If a user wants to connect via CLI, they have to add `core-tools` layer to their provision configuration.
```

So IIUC, if the `secure-management` is provided, then the `jdr.sh` will use the secure interface to connect the server, right? Is the communication happening over HTTP(S)?

Reply:
You can choose via the `jdr.sh` arguments the communication protocol/port/host. By default it uses `http+remoting`

Reply:
Ok, thanks.